### PR TITLE
Enforce CSRF protection for login view

### PR DESCRIPTION
**Pull request by cutwater**

### CODE REVIEW

**Comment by cutwater** [comment]
@newswangerd You to GET request to the `.../auth/login` endpoint.

File: galaxy_ng/app/api/ui/views/auth.py [hunk 1]
```diff
@@ -19,10 +22,24 @@
...
+        return user, None
+
+
 class LoginView(api_base.GenericAPIView):
     serializer_class = LoginSerializer
+    authentication_classes = (_CSRFSessionAuthentication,)
     permission_classes = (RestrictOnCloudDeployments,)
 
+    @method_decorator(ensure_csrf_cookie)
+    def get(self, request, *args, **kwargs):
```
Comment (alikins) [inline galaxy_ng/app/api/ui/views/auth.py:40]: To verify I understand correctly, the _CSRFSessionAuthentication Authentication class is used to enfore existence of a csrf token, even for unauthenticated or "safe" operations.

The 'get' method here requires user, auth, and csrf token and returns an empty response body. But the ensure_csrf_cookie() decorator will set the right response headers to include any needed fields for csrf
(either in the django session, or a csrf token in a cookie).

**Comment by cutwater** [comment]
@alikins it isn't about requiring CSRF for "safe" methods. It's about requiring CSRF when no `sessionid` cookie is set.

File: galaxy_ng/app/api/ui/views/auth.py (hunk 1 above)
Comment (cutwater) [reply galaxy_ng/app/api/ui/views/auth.py:40]: See my comment below. For unauthenticated and unsafe. Specifically for `POST` method to `.../auth/login`.

**Pull request merged**